akka-http-tests/src/test/java/akka/http/javadsl/server/directives/HeaderDirectivesTest.java

```diff
@@ -153,11 +153,11 @@ public void testOptionalHeaderValueByName() {
 
   @Test
   public void testOptionalHeaderValueByType() {
-    TestRoute route = testRoute(optionalHeaderValueByType(Server.class,
-      (Optional<Server> s) -> complete(((Boolean)s.isPresent()).toString())));
+    TestRoute route = testRoute(optionalHeaderValueByType(UserAgent.class,
+      (Optional<UserAgent> ua) -> complete(((Boolean)ua.isPresent()).toString())));
 
     route
-      .run(HttpRequest.create().addHeader(Server.create(ProductVersion.create("such-service", "0.6"))))
+      .run(HttpRequest.create().addHeader(UserAgent.create("custom-server/1.2.3")))
       .assertStatusCode(StatusCodes.OK)
       .assertEntity("true");
```

Comment:
should this rather be an additional test rather than changing this one?

Reply:
`Server` header was already used in another test above. Therefore I shoehorned `UserAgent` factory usage into this one.